js/core.js

```diff
@@ -869,3 +886,24 @@ function sleep(duration) {
         setTimeout(function() { resolve(); }, duration);
     });
 }
+
+
+class OptionalPermissions {
```

Comment:
This class would be useful for other cases too, like context menu. Maybe add a TODO comment to implement it to other areas later.

Reply:
I was actually going to remove this class, as I'll be changing the permissions from optional to required.